pkg/scheduler/service/service.go

```diff
@@ -224,21 +226,30 @@ func (s *SchedulerServer) AddHyperkubeFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.KMPath, "km_path", s.KMPath, "Location of the km executable, may be a URI or an absolute file path.")
 }
 
-// returns (downloadURI, basename(path))
 func (s *SchedulerServer) serveFrameworkArtifact(path string) (string, string) {
+	return s.serveFrameworkArtifactWithFilename(path, "")
+}
+
+// returns (downloadURI, basename(path))
```

Comment:
You mean compute the basename ? I was handling the case of an empty basename inside serveFrameworkArtifactWithFilename, but you're right it is better to do that here. In my next pull request I also removed the second return parameter in serveFrameworkArtifactWithFilename now that filename is systematically used (no edge case).